### PR TITLE
Fix/artflt 1292 fix facebook advertisertrackingenabled flag after flutter upgrade

### PR DESCRIPTION
**Pull request by sarahemaginnis**
## Summary
Added explicit AdvertiserTrackingEnabled flag setting to fix Facebook iOS attribution issues after Flutter 3.27->3.32.5 upgrade.

## Changes Made

**ADDED** `setAdvertiserTrackingEnabled()` method to `lib/flutter_facebook_app_links.dart`:
```dart
/// Sets the advertiser tracking enabled status for Facebook SDK.
///
/// This method controls whether the Facebook SDK reports advertiser tracking
/// as enabled to Facebook servers. Call this after obtaining ATT permission
/// on iOS and before calling consentProvided() or consentRevoked().
///
/// On iOS: Controls Settings.shared.isAdvertiserTrackingEnabled
/// On Android: Controls FacebookSdk.setAdvertiserIDCollectionEnabled
///
/// [enabled] - Whether advertiser tracking should be enabled based on user consent
static Future<void> setAdvertiserTrackingEnabled(bool enabled) async {
  await _channel.invokeMethod('setAdvertiserTrackingEnabled', {'enabled': enabled});
}
```

### iOS (ios/Classes/SwiftFlutterFacebookAppLinksPlugin.swift)
Added "setAdvertiserTrackingEnabled" case to handle method:
```swift
case "setAdvertiserTrackingEnabled":
    if let arguments = call.arguments as? [String: Any],
       let enabled = arguments["enabled"] as? Bool {
        Settings.shared.isAdvertiserTrackingEnabled = enabled
        result(nil)
    } else {
        result(FlutterError(code: "INVALID_ARGUMENTS",
                          message: "Expected boolean 'enabled' parameter",
                          details: nil))
    }
```

### Android (android/src/main/java/it/remedia/flutter_facebook_app_links/FlutterFacebookAppLinksPlugin.java)
Added "setAdvertiserTrackingEnabled" case to onMethodCall:
```java
else if (call.method.equals("setAdvertiserTrackingEnabled")) {
    Boolean enabled = call.argument("enabled");
    if (enabled != null) {
        FacebookSdk.setAdvertiserIDCollectionEnabled(enabled);
        result.success(null);
    } else {
        result.error("INVALID_ARGUMENTS", "Expected boolean 'enabled' parameter", null);
    }
}
```

### 📚 Documentation Updates (README.md)
**COMPLETELY REWROTE** the consent/consentRevoked sections with:

- **Correct call order**: ATT permission → setAdvertiserTrackingEnabled() → consentProvided()
- Platform-specific method explanations:
  - **iOS**: Settings.shared.isAdvertiserTrackingEnabled (Facebook tracking reporting)
  - **Android**: FacebookSdk.setAdvertiserIDCollectionEnabled (advertiser ID collection)
## Root Cause
After Flutter 3.27→3.32.5 upgrade, Facebook SDK's AdvertiserTrackingEnabled flag wasn't being set properly due to ATT (App Tracking Transparency) changes. The existing `consentProvided()`/`consentRevoked()` calls were insufficient - explicit flag setting is required.

## API Method Differences & Semantics
- **iOS**: `setAdvertiserTrackingEnabled` controls `Settings.shared.isAdvertiserTrackingEnabled` (reports tracking status to Facebook)
- **Android**: `setAdvertiserTrackingEnabled` controls `FacebookSdk.setAdvertiserIDCollectionEnabled` (controls advertiser ID collection)

Both affect Facebook's attribution system but have platform-specific semantics.

## Recommended Usage Pattern
```dart
// iOS with ATT permission
if (Platform.isIOS) {
  // Request ATTrackingManager permission first
  final status = await AppTrackingTransparency.requestTrackingAuthorization();
  final trackingEnabled = status == TrackingStatus.authorized;

  await FlutterFacebookAppLinks.setAdvertiserTrackingEnabled(trackingEnabled);
  await FlutterFacebookAppLinks.consentProvided(); // or consentRevoked()
}

// Android (no ATT equivalent)
await FlutterFacebookAppLinks.setAdvertiserTrackingEnabled(userConsentGiven);
await FlutterFacebookAppLinks.consentProvided(); // or consentRevoked()
```

## Expected Outcome
- Facebook Events Manager will show `"AdvertiserTrackingEnabled": "true"` for consenting users
- iOS campaign attribution will work correctly
- StartTrial events will appear in Events Manager again
- Proper GDPR compliance maintained

## Testing Steps
1. **Build & Deploy**: Push to flutter_facebook_app_links repo, update main app dependency
2. **Clean Build**: `fvm flutter clean && fvm flutter pub get && fvm flutter build ios --debug`
3. **iOS ATT Testing**: Test on physical iOS device with ATT permission flows
4. **Verification**: Check Facebook Events Manager shows correct AdvertiserTrackingEnabled values

## Files Modified
- ✅ `ios/Classes/SwiftFlutterFacebookAppLinksPlugin.swift` - iOS native method
- ✅ `android/src/main/java/it/remedia/flutter_facebook_app_links/FlutterFacebookAppLinksPlugin.java` - Android native method
- ✅ `lib/flutter_facebook_app_links.dart` - Dart API method
- ✅ `README.md` - Complete documentation & usage patterns

This implementation resolves the critical Facebook attribution issue and provides a complete, well-documented API for proper tracking consent management.

## Documentation
[https://developers.facebook.com/docs/app-events/guides/advertising-tracking-enabled/](https://developers.facebook.com/docs/app-events/guides/advertising-tracking-enabled/)

### CODE REVIEW

**Comment by twinsundev** [comment]
### 🤖 Scarif Code Review

## Summary
This PR adds a new method `setAdvertiserTrackingEnabled` to both iOS and Android native implementations to allow explicit control over Facebook's advertiser tracking flag. The implementation addresses iOS attribution issues that occurred after upgrading Flutter from 3.27 to 3.32.5.

**Positive aspects:**
- Clear problem statement and well-documented issue in the PR description
- Consistent implementation pattern across both platforms (iOS and Android)
- Proper error handling for invalid arguments
- Follows existing code style and conventions in the plugin

## Priority Issues

### 🔴 CRITICAL: Missing Dart API exposure
The new `setAdvertiserTrackingEnabled` method is implemented in the native code (iOS and Android) but is **not exposed in the Dart API** (`lib/flutter_facebook_app_links.dart`). This means developers cannot actually call this method from their Flutter code.

**Required action:** Add a static method to `FlutterFacebookAppLinks` class:
```dart
static Future<void> setAdvertiserTrackingEnabled(bool enabled) async {
  await _channel.invokeMethod('setAdvertiserTrackingEnabled', {'enabled': enabled});
}
```

Without this, the native implementation is unreachable from Flutter apps.

### 🟡 MEDIUM: Missing documentation
The new method should be documented in the README.md to help users understand:
- When to call this method
- How it relates to ATT (App Tracking Transparency) permission flow
- The relationship between this method and the existing `consentProvided()`/`consentRevoked()` methods
- Example usage in the context of obtaining ATT permission

Currently, line 78 of README.md mentions calling `consentProvided()` after obtaining ATT permission, but doesn't mention the new `setAdvertiserTrackingEnabled()` method.

## Suggestions

### 1. Consider API naming consistency
The Android implementation uses `FacebookSdk.setAdvertiserIDCollectionEnabled()` while iOS uses `Settings.shared.isAdvertiserTrackingEnabled`. Both are correct for their respective platforms, but it's worth noting that:
- Android controls "Advertiser ID Collection" 
- iOS controls "Advertiser Tracking Enabled"

These are semantically similar but not identical. Consider adding a comment in the code or documentation explaining this subtle difference.

### 2. Method call timing
Consider documenting the recommended order of operations:
1. Request ATT permission (using iOS ATTrackingManager)
2. Call `setAdvertiserTrackingEnabled(true/false)` based on user consent
3. Call `consentProvided()` or `consentRevoked()` to initialize the SDK

The relationship between these methods could be clearer.

### 3. Testing considerations
The PR description mentions testing steps, but these focus on manual testing. Consider adding:
- Unit tests for the new method (if the plugin has a test suite)
- Edge case handling: what happens if this is called before SDK initialization?

## Security Assessment
✅ **No security issues identified**
- Input validation is properly implemented for the boolean parameter
- Error handling returns appropriate error codes
- No sensitive data exposure concerns
- Method follows principle of least privilege (only controls one specific flag)

## Additional Notes

### Code Quality
The implementation is clean and follows the existing patterns in the codebase:
- Matches the structure of other method handlers
- Consistent error codes ("INVALID_ARGUMENTS")
- Proper null checking
- Appropriate return values (null on success)

### Android-specific observation
Line 86: `FacebookSdk.setAdvertiserIDCollectionEnabled(enabled)` is the correct Android SDK method. Note that this was introduced in Facebook SDK v5.9.0+. Consider checking if the `android/build.gradle` specifies a compatible SDK version.

### iOS-specific observation  
Line 32: `Settings.shared.isAdvertiserTrackingEnabled = enabled` correctly sets the Facebook SDK setting. This is separate from iOS's ATTrackingManager permission and should be called after obtaining ATT authorization status.

## Recommendation
**Request Changes** - The Dart API must be added before this can be merged. Once the Dart method is added and documentation is updated, this PR will be ready to merge.

---
*Review generated by Scarif AI Code Review*

File: ios/Classes/SwiftFlutterFacebookAppLinksPlugin.swift [hunk 1]
```diff
@@ -26,6 +26,16 @@ public class SwiftFlutterFacebookAppLinksPlugin: NSObject, FlutterPlugin {
       Settings.shared.isAutoLogAppEventsEnabled = false
       ApplicationDelegate.shared.initializeSDK()
       result(nil)
+    case "setAdvertiserTrackingEnabled":
```
Comment (twinsundev) [inline ios/Classes/SwiftFlutterFacebookAppLinksPlugin.swift]: Good error handling! The implementation correctly validates the argument and returns a descriptive error if the `enabled` parameter is missing or not a boolean.

Minor suggestion: Consider whether this should also handle the case where the SDK is not yet initialized. Currently, calling this before `initFBLinks()` or `consentProvided()` might have unexpected behavior.

File: android/src/main/java/it/remedia/flutter_facebook_app_links/FlutterFacebookAppLinksPlugin.java [hunk 2]
```diff
@@ -80,6 +80,14 @@ public void onMethodCall(MethodCall call, Result result) {
...
       FacebookSdk.fullyInitialize();
       result.success("");
+    } else if (call.method.equals("setAdvertiserTrackingEnabled")) {
+      Boolean enabled = call.argument("enabled");
+      if (enabled != null) {
+        FacebookSdk.setAdvertiserIDCollectionEnabled(enabled);
+        result.success(null);
+      } else {
+        result.error("INVALID_ARGUMENTS", "Expected boolean 'enabled' parameter", null);
+      }
```
[the remark this answers is not in the document]
Comment (sarahemaginnis) [reply android/src/main/java/it/remedia/flutter_facebook_app_links/FlutterFacebookAppLinksPlugin.java]: Error handling is already aligned and identical.

**Review by fbcouch — APPROVED** [review]
